src/main/java/org/osc/controller/nsc/api/SampleSdnRedirectionApi.java

```diff
@@ -411,6 +427,23 @@ public NetworkElement updateNetworkElement(NetworkElement portGroup, List<Networ
         });
     }
 
+    public NetworkElement updateNetworkElement(NetworkElement nwElement) throws Exception {
```

Comment:
please name these methods distinctly as well. This is basically creating a port even though internally we are creating a Network Element. So basically this method could be createPort, updatePort etc

In NSC context, *NetworkElement methods refer to port group(in neutron sfc they refer to PPG's). Adding yet another overload on top of *NetworkElement method seems confusing.

Reply:
@arvindn05 , I see an issue because in the sdn-controller-api sdk, I already see API's like getNetworkElements, etc. So renaming just the support methods like createPorts, getPorts etc will lead to confusion. So i plan to keep it as is and refine the methods names where needed (like some comments that is applicable).

Reply:
its much harder to see the distinction between the support methods and the sdk methods.

updateNetworkElement(2  params) (SDK method) -> updateNetworkElement (single param)

work completely differently. SDK referes to updating port group and updateNetworkElement refers to updating a port.

@emanoelxavier your thougths?

Reply:
Currently the consistency pattern is entity name (networkelemententity) -> rest api name (network element resource). The question here is about how to name the helper methods in between.  I like the idea of calling it `port` however if we are doing that then the entity should be `portentity` and the rest resource `port` instead of `network element`. In this case the helper methods then could be called `port`. This would make it all consistent across the board. But this is also a bigger change since we would rename the tables and entities as well. My recommendation would be rename the helper methods as arvind mention to indicate that intent (these things are really ports) and place a TODO on the entity and rest api to have it renamed to port as well in the future. Makes sense?

Reply:
@emanoelxavier, I will implement this way and put todo for the entity and the rest api's.